index.js

```diff
@@ -91,6 +100,28 @@ function _insertCredentials(username, userhash) {
 let cachedAgents;
 
 /**
+ * Generates and maps an internal username to the given username
+ *
+ * To protect against username collisions, this function takes
+ * a passed in username and adds a hash to it, generating an
+ * internal username.
+ *
+ * @param {string} username Username of the user.
+ *
+ * @return {string} Internal username
+ *
+ * @private
+ */
+function _generateInternalUsername(username) {
```

Comment:
I reversed the previous condition and encapsulated the logic inside this private method, compared to the previous: https://github.com/sugarcrm/thorn/pull/122/files#diff-168726dbe96b3ce427e7fedce31bb0bcR340